raiden/messages.py

```diff
@@ -1700,6 +1704,112 @@ def verify_request_monitoring(
         )
 
 
+class UpdatePFS(SignedMessage):
+    cmdid = messages.UPDATEPFS
+    """ Message to inform a pathfinding service about a capacity change. """
+
+    def __init__(
+            self,
+            nonce: typing.Nonce,
+            transferred_amount: typing.TokenAmount,
+            locked_amount: typing.TokenAmount,
+            locksroot: typing.Locksroot,
+            token_network_address: typing.TokenNetworkAddress,
+            channel_identifier: typing.ChannelID,
+            chain_id: typing.ChainID,
+            reveal_timeout: int,
+            signature: typing.Optional[typing.Signature] = None,
+    ) -> None:
+        self.chain_id = chain_id
+        self.nonce = nonce
+        self.transferred_amount = transferred_amount
+        self.locked_amount = locked_amount
+        self.locksroot = locksroot
+        self.channel_identifier = channel_identifier
+        self.token_network_address = token_network_address
+        self.reveal_timeout = reveal_timeout
+        if signature is None:
+            self.signature = b''
+        else:
+            self.signature = signature
+
+    @classmethod
+    def from_balance_proof(
+            cls: typing.Type[typing.T_UpdatePFS],
+            balance_proof: BalanceProofUnsignedState,
+            reveal_timeout: int,
+    ) -> typing.T_UpdatePFS:
+        assert isinstance(balance_proof, BalanceProofUnsignedState)
+        return cls(
+            nonce=balance_proof.nonce,
+            transferred_amount=balance_proof.transferred_amount,
+            locked_amount=balance_proof.locked_amount,
+            locksroot=balance_proof.locksroot,
+            token_network_address=balance_proof.token_network_identifier,
```

Comment:
For now, the values are pretty much the same but i agree that it could be confusing. We should change `token_network_address` to `token_network_identifier` in this case.

Reply:
so...the issue is, that
-  `NettingChannelState` calls it `_identifier`, although it is initiated just from the `_address`: https://github.com/raiden-network/raiden/blob/d48396c071d14e41b78cc9de3e811c9ac85a49ad/raiden/blockchain/state.py#L60
- this is then reused when creating `BalanceProofUnsignedState` instances, e.g. https://github.com/raiden-network/raiden/blob/d48396c071d14e41b78cc9de3e811c9ac85a49ad/raiden/transfer/channel.py#L1185
- most (if not all) other off-chain messages are just using `token_network_address`.

So, while it might look more consistent in scope of this factory to use `token_network_identifier`, it is actually inconsistent with the other messages defined in this module (also with the spec).

See also (and contribute to) the discussion at: https://github.com/raiden-network/raiden/issues/3493